File: beginner_tutorials/scripts/beginner_tutorials.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import String

def callback(data):
    rospy.loginfo(rospy.get_caller_id() + "I heard %s", data.data)

if __name__ == '__main__':
    rospy.init_node('beginner_tutorials', anonymous=False)
    pub = rospy.Publisher('chatter', String, queue_size=10)
    rospy.Subscriber("chatter", String, callback)
    rate = rospy.Rate(10)

    while not rospy.is_shutdown():
        hello_str = "hello world! %s" % rospy.get_time()
        pub.publish(hello_str)
        rospy.loginfo(hello_str)
        rate.sleep()

File: beginner_tutorials/src/beginner_tutorials.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

void callback(const std_msgs::String::ConstPtr& msg)
{
  ROS_INFO("I heard: %s", msg->data.c_str());
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "beginner_tutorials");
  ros::NodeHandle nh;

  ros::Publisher pub = nh.advertise<std_msgs::String>("chatter", 10);
  ros::Subscriber sub = nh.subscribe("chatter", 10, callback);
  ros::Rate rate(10);

  while (ros::ok())
  {
    std_msgs::String msg;
    std::stringstream ss;
    ss << "hello world! " << ros::Time::now().toSec();
    msg.data = ss.str();

    pub.publish(msg);
    ROS_INFO("%s", msg.data.c_str());

    ros::spinOnce();
    rate.sleep();
  }

  return 0;
}
